src/main.vue

```diff
@@ -0,0 +1,418 @@
+                    }
+                    return val;
+                };
+            },
+        },
+    });
+
+    let defaultLang = state.setting('language');
+    let preferredLangs = _.clone(window.navigator && window.navigator.languages) || [];
+
+    // our configs default lang overrides all others
+    if (defaultLang) {
+        preferredLangs.unshift(defaultLang);
+    }
+
+    // set a default language
+    i18next.changeLanguage('en-us');
+
+    // Go through our browser languages until we find one that we support
+    for (let idx = 0; idx < preferredLangs.length; idx++) {
+        let lang = preferredLangs[idx];
+
+        // if this is a language such as 'fr', add a following one of 'fr-fr' to cover
+        // both cases
+        if (lang.length === 2) {
+            preferredLangs.splice(idx + 1, 0, lang + '-' + lang);
+        }
+
+        if (_.includes(AvailableLocales.locales, lang.toLowerCase())) {
+            i18next.changeLanguage(lang, (err, t) => {
+                if (err) {
+                    // setting the language failed so set default again
+                    i18next.changeLanguage('en-us');
+                }
+            });
+            break;
+        }
+    }
+}
+
+async function initState() {
+    let stateKey = state.settings.startupOptions.state_key;
+
+    // Default to a preset key if it wasn't set
+    if (typeof stateKey === 'undefined') {
+        stateKey = 'kiwi-state';
+    }
+
+    let persistLog = Logger.namespace('StatePersistence');
+    let persist = new StatePersistence(stateKey || '', state, Storage, persistLog);
+    persist.includeBuffers = !!state.settings.startupOptions.remember_buffers;
+
+    if (stateKey) {
+        await persist.loadStateIfExists();
+    }
+
+    api.setState(state);
+}
+
+function initThemes() {
+    let themeMgr = ThemeManager.instance(state);
+    api.setThemeManager(themeMgr);
+
+    let argTheme = getQueryVariable('theme');
+    if (argTheme) {
+        themeMgr.setTheme(argTheme);
+    }
+}
+
+function initSound() {
+    let sound = new SoundBleep();
+    let bleep = new AudioManager(sound);
+
+    bleep.listen(state);
+    bleep.listenForHighlights(state);
+}
+
+function initInputCommands() {
+    /* eslint-disable no-new */
+    new InputHandler(state);
+}
+
+function startApp() {
+    api.emit('init');
+
+    /* eslint-disable no-new */
+    new Vue({
+        el: '#kiwi-app',
+        render: h => h(App),
+        i18n: new VueI18Next(i18next),
+    });
+
+    api.emit('ready');
+}
+
+function showError(err) {
+    if (err) {
+        log.error('Error starting Kiwi IRC:', err);
+    } else {
+        log.error('Unknown error starting Kiwi IRC');
+    }
+
+    /* eslint-disable no-new */
+    new Vue({
+        el: '#kiwi-app',
+        render: h => h(StartupError),
+    });
+}
+export default{
+    props: ['kiwiConfig'],
+    mounted() {
+        window.kiwiConfig = () => this.kiwiConfig;
```

Comment:
It looks like it's reading the config from the components `kiwiConfig` prop, but then it's setting that to `window.kiwiConfig()` and that's where it's read from in the kiwi app? Why not just use the property directly?